### DISCUSSION TITLE
iperf3 stop running after 300 sec with error - unable to send control message : Broken pipe

### DISCUSSION THREAD

**Discussion by initChat** [esnet/iperf#1499] [Q&A]
Please help me, I have a problem of IPerf3 stop running after 300 sec (5 mins), when I want to use Iperf3 for testing for at least 24 hours. The error said **iperf3: error - unable to send control message: Broken pipe**

I use IPerf3 version 3.9 on ubuntu of windows 10 wsl for client, and the server side is the public domain that I don't have access to. So I can only control the client side. The client is connect to WiFi from CPE that inserted 5G sim card, while the server is on another network but it is a public domain and has configured iperf3 enable on the port 5021.

The client code on my side is run as below.

`iperf3 -c <IP address> -i 10 -t 1500`

and get the result as
```
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-60.00  sec  28.5 MBytes  3.99 Mbits/sec  338   10.8 KBytes       
[  5]  60.00-120.00 sec  27.9 MBytes  3.90 Mbits/sec  229   16.3 KBytes       
[  5] 120.00-180.00 sec  28.1 MBytes  3.92 Mbits/sec  255   14.9 KBytes       
[  5] 180.00-240.00 sec  27.9 MBytes  3.90 Mbits/sec  228   9.49 KBytes       
[  5] 240.00-300.00 sec  27.9 MBytes  3.90 Mbits/sec  225   16.3 KBytes       
[  5] 240.00-300.00 sec  27.9 MBytes  3.90 Mbits/sec  225   16.3 KBytes       
iperf3: error - unable to send control message: Broken pipe
```
<please click on the inserted link, as I don't know how to attach pictures>
and i try --debug and show result in excel in this [screenshot](https://ibb.co/m5W4NgV) which is the snippet from 240 sec - 300 sec. And you can see that at 290-300 sec there is no tcpi_snd_cwnd sending, same with the error that said unable to send control message.

**What should I do to make the iperf3 not stop running?**

Please helps me, I have searched multiple places in the internet and don't encounter anyone that has experience the same issue. Thank you very much in advance.

**My current suspicious**
- from this [link](https://github.com/esnet/iperf/issues/1069), they have mentioned in this [screenshot](https://ibb.co/F3KHqjH) that broken pipe could be the reason from memory full. But I still don't know how to fix this issue.
- from this [link](https://github.com/esnet/iperf/discussions/1296) mentioned that run multiple UE streams to the iperf could case memory full or memory leakage.

**What I have tried so far**, which still have the same problem. The things I have tried can be irrelevant as I am so new to Iperf.
- change from computer to be Raspberry pi -> still get the same problem
- use -N (no delay), increase windows size (-w), change from -t to -n 1024M -l 32k
- try parallel stream

**Comment by davidBar-On** [comment]
Since the issue seem to happen exactly after 5 minutes, my guess is that the public domain server limits the connections to 5 minutes.  Suggest to check that with the public domain owners/documentation.  (By the way, the public domain IP address is shown in the results you included - you may edit you comment if this is not desired.)

**Reply by initChat** [reply]
Thank you for replying. I will contact the server and update about the result. And thank you for reminding about the server showing, i edit it, thank you very much.
